### ISSUE TITLE
Conflict on old libffi version

### ISSUE THREAD

**Issue by ppodgorsek** [docker-library/python#469]
When using python:3.8-alpine3.11 as the base image and extending it by installing Firefox 75.0 (requiring libffi 3.3), the build fails with the following error:

> ERROR: unsatisfiable constraints:
>
>   libffi-3.3-r1:
>     conflicts: libffi-3.2.1-r6
>     satisfies: libffi-dev-3.3-r1[libffi=3.3-r1]
>
>   libffi-3.2.1-r6:
>     conflicts: libffi-3.3-r1
>     breaks: libffi-dev-3.3-r1[libffi=3.3-r1]
>     satisfies: .python-rundeps-20200324.025755[so:libffi.so.6]

When forcing the version to libffi 3.2.1, I get the opposite issue, with Firefox needing a more recent version of libffi.

This used to work for older versions of Firefox on python:3.7-alpine3.10.

Would it be possible to upgrade the version of libffi in the Python image please? From what I see in the Dockerfile, it would probably only require rebuilding the image and publishing it, with no code changes.

**Comment by tianon** [comment]
Just rebuilding won't help because we'll still get `libffi-3.2.1-r6` -- that's the latest available in Alpine 3.11.

When trying to install `firefox` from edge, I get more issues than just this one, though:

```console
/ # apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/ma
in --repository http://dl-cdn.alpinelinux.org/alpine/edge/community firefox
fetch http://dl-cdn.alpinelinux.org/alpine/edge/community/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/edge/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  libffi-3.2.1-r6:
    conflicts: libffi-3.3-r1
    satisfies: libffi-dev-3.2.1-r6[libffi=3.2.1-r6]
               firefox-esr-68.7.0-r0[so:libffi.so.6]
               glib-2.62.6-r0[so:libffi.so.6]
               wayland-libs-client-1.17.0-r0[so:libffi.so.6]
               p11-kit-0.23.18.1-r0[so:libffi.so.6]
               .python-rundeps-20200324.025755[so:libffi.so.6]
  libffi-3.3-r1:
    conflicts: libffi-3.2.1-r6
    breaks: libffi-dev-3.2.1-r6[libffi=3.2.1-r6]
    satisfies: firefox-75.0-r1[so:libffi.so.7]
  icu-libs-66.1-r0:
    conflicts: icu-libs-64.2-r1
    satisfies: firefox-75.0-r1[so:libicui18n.so.66]
               firefox-75.0-r1[so:libicuuc.so.66]
  firefox-75.0-r1:
    conflicts: firefox-esr-68.7.0-r0[so:liblgpllibs.so=0]
               firefox-esr-68.7.0-r0[so:libmozavcodec.so=0]
               firefox-esr-68.7.0-r0[so:libmozavutil.so=0]
               firefox-esr-68.7.0-r0[so:libmozgtk.so=0]
               firefox-esr-68.7.0-r0[so:libmozsandbox.so=0]
               firefox-esr-68.7.0-r0[so:libmozwayland.so=0]
               firefox-esr-68.7.0-r0[so:libxul.so=0]
    satisfies: world[firefox]
  icu-libs-64.2-r1:
    conflicts: icu-libs-66.1-r0
    satisfies: firefox-esr-68.7.0-r0[so:libicui18n.so.64]
               firefox-esr-68.7.0-r0[so:libicuuc.so.64]
  firefox-esr-68.7.0-r0:
    conflicts: firefox-75.0-r1[so:liblgpllibs.so=0]
               firefox-75.0-r1[so:libmozavcodec.so=0]
               firefox-75.0-r1[so:libmozavutil.so=0]
               firefox-75.0-r1[so:libmozgtk.so=0]
               firefox-75.0-r1[so:libmozsandbox.so=0]
               firefox-75.0-r1[so:libmozwayland.so=0]
               firefox-75.0-r1[so:libxul.so=0]
    satisfies: world[firefox-esr]
```

(Edge and 3.11 aren't designed to mix cleanly, AFAIK -- you might have better luck taking our `Dockerfile` and switching the `FROM` directly to `alpine:edge`.)

**Comment by ppodgorsek** [comment]
Indeed, it was a mistake in our Dockerfile. I changed the repos to the standard Alpine 3.11 ones with Firefox ESR and everything is back to normal. Thanks for pointing out the mistake!
